subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/catalog/TomlDependenciesExtensionIntegrationTest.groovy

```diff
@@ -185,6 +185,44 @@ myBundle = ["lib", "lib2"]
         }
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/22552")
+    void "can add several dependencies at once using a bundle with DependencyHandler#addProvider"() {
+        tomlFile << """[libraries]
+lib = {group = "org.gradle.test", name="lib", version.require="1.0"}
+lib2.module = "org.gradle.test:lib2"
+lib2.version = "1.0"
+
+[bundles]
+myBundle = ["lib", "lib2"]
+"""
+        def lib = mavenHttpRepo.module("org.gradle.test", "lib", "1.0").publish()
+        def lib2 = mavenHttpRepo.module("org.gradle.test", "lib2", "1.0").publish()
+        buildFile << """
+            apply plugin: 'java-library'
+
+            dependencies {
+                addProvider("implementation", libs.bundles.myBundle)
+            }
+        """
+
+        when:
```

Comment:
Very minor, this seems like it should be removed and the first `then` below be made `when`.